### PR TITLE
NodeDefaults object for 'default' settings on different networks

### PR DESCRIPTION
**Pull request by codingupastorm**
- Groups all Setting defaults into one object inside Network. AKA all the values that will be selected if no arguments are passed in at the command-line.
- Avoids live networks "invisibly" functioning differently to test via `RequireStandard`.
- Will be built upon for different mempool defaults across networks. See #3019

### CODE REVIEW

File: src/NBitcoin/Network.cs [hunk 1]
```diff
@@ -107,14 +107,9 @@ public abstract class Network
         public int DefaultPort { get; protected set; }
```
Comment (codingupastorm) [inline src/NBitcoin/Network.cs]: Thought about putting DefaultPort and RPCPort in NodeDefaults but decided against - it seems they're not really defaults but actual network settings, aka you can't really change them, they're used too prolifically

**Comment by quantumagi** [comment]
This is not an approach that can be extended to all settings objects due to the add-in nature of features.

A simpler approach will provide the required visibility whilst remaining **feature-modular**:

```
private const int TEST_REQUIRES_STANDARD = true;
private const int NON_TEST_REQUIRES_STANDARD = false;
...
this.RequireStandard = !(config.GetOrDefault("acceptnonstdtxn", nodeSettings.Network.IsTest() ? TEST_REQUIRES_STANDARD : NON_TEST_REQUIRES_STANDARD, this.logger));
```

**Comment by codingupastorm** [comment]
Thanks @zeptin I flipped the settings around to be AcceptNonStandardTransaction, just like the cmd line option is. Those tests are passing locally now.

**Comment by codingupastorm** [comment]
@quantumagi I think the mempool will likely not exist as a feature and be a core part of the node in the future.  For cases where 3rd-party features are injected devs will be creating their own networks anyway, so they can cast to their network type to access the required settings if needed. (This happens with the FederatedPeg networks in a few places already I believe.

The logic you've given might not fix the problem - we are trying to avoid any situations where test and live networks behave differently implicitly, and it doesn't cater for things like 'DefaultAncestorSizeLimit' etc in the future.

**Comment by quantumagi** [comment]
Re: "The logic you've given might not fix the problem - we are trying to avoid any situations where test and live networks behave differently implicitly, and it doesn't cater for things like 'DefaultAncestorSizeLimit' etc in the future.". 

What is happening in the Bitcoin Main and Test networks classes is really no different and we are breaking the model for no real benefit. We are instead simply reshuffling the code by adding some extra class and moving the provisioning of defaults to the network classes. 

The main point of the current and suggested logic is that the feature settings classes should be in control of both the available settings and the defaults thereof to retain the add-in nature of features.

**Comment by codingupastorm** [comment]
>The main point of the current and suggested logic is that the feature settings classes should be in control of both the available settings and the defaults thereof to retain the add-in nature of features.

Right, this is the assumption I'm *trying* to break. On different networks, the defaults will be different. The features shouldn't control this. The network should

**Comment by quantumagi** [comment]
I am gathering from all of this that you believe you should be able to look at the network class and see all the defaults right there. The only problem is that it will never be fully possible and we now end up with a hybrid approach.

**Comment by bokobza** [comment]
I find it strange to have some network properties inside an object called NodeDefaults.
Also trying to understand what we're trying to achieve here.
Is it a way to give people creating their own network a way to set up defaults properties more easily?

**Comment by codingupastorm** [comment]
Different networks want different default settings for things like ancestor size limits etc. Currently the defaults are hardcoded and can't be different per network.

What we achieve is allowing networks to have different settings for node default behaviour for things like -acceptnonstdtxn, -ancestorsizelimit etc.

**Comment by noescape00** [comment]
@zeptin @rowandh @quantumagi @bokobza @codingupastorm what is the status of this PR?

**Comment by noescape00** [comment]
Closing since PR is abandoned. 
Please reopen if needed

**Pull request closed without merging**